### PR TITLE
Yki 114 exam session listing

### PR DESCRIPTION
**Pull request by matiisi**
This unusually large PR touches `registration` in the following ways:

- added mock data for local development 
- exam sessions are fetched on user activity
  - selecting language after service description or
  - changing filters in the listing view
- exam session data is parsed and stored in redux state for easy access
- created new components for listing exam sessions
- exam sessions are grouped by exam date and listed

also:

- updated dependencies to latest versions

### CODE REVIEW

File: src/main/js/src/store/actions/registration.js [hunk 1]
```diff
@@ -1,66 +1,96 @@
...
-  const locations = ['Pernäjä', 'Hämeenlinna', 'Jämsä', 'Korso', 'Ankkalinna'];
+export const fetchExamSessions = () => {
   return dispatch => {
-    dispatch(fetchExamLocationsStart());
-    // TODO: get actual locations from api
-    // axios.get('/yki/api/exam-sessions')
-    dispatch(fetchExamLocationsSuccess(locations));
+    dispatch(fetchExamSessionsStart());
+    axios
+      .get('/yki/api/exam-sessions')
```
Comment (tbertell) [inline src/main/js/src/store/actions/registration.js]: endpoint is /yki/api/exam-session

File: src/main/js/src/store/reducers/registration.js [hunk 2]
```diff
@@ -12,23 +15,54 @@ const initialState = {
...
+        examSessions: action.examSessions,
       };
-    case actionTypes.FETCH_EXAM_LOCATIONS_FAIL:
+    case actionTypes.FETCH_EXAM_SESSIONS_FAIL:
       return {
         ...state,
         loading: false,
         error: action.error,
       };
+    case actionTypes.FILTER_AND_GROUP_BY_DATE:
```
Comment (tbertell) [inline src/main/js/src/store/reducers/registration.js]: Consider using ramda for data transformations. For example group by session date would be:
R.groupBy(s => s.session_date, filteredByLanguage)

File: src/main/js/src/store/actions/registration.js (hunk 1 above)
Comment (matiisi) [reply src/main/js/src/store/actions/registration.js]: Oops! Will fix on next branch while improving visuals and hiding exam sessions in collapsible element.

File: src/main/js/src/store/reducers/registration.js (hunk 2 above)
Comment (matiisi) [reply src/main/js/src/store/reducers/registration.js]: You are totally correct.
I need to start looking into Ramda and find out what it can do.

**Pull request merged**